lib/vector-common/src/internal_event/mod.rs

```diff
@@ -248,23 +248,38 @@ macro_rules! registered_event {
     };
 }
 
-#[derive(Clone)]
-pub struct Cached<Tags, Event, Register> {
-    cache: Arc<RwLock<BTreeMap<Tags, Event>>>,
-    register: Register,
+pub trait RegisterEvent<Tags, Event>
+where
+    Event: RegisterInternalEvent,
+{
+    fn register(tags: &Tags) -> <Event as RegisterInternalEvent>::Handle;
+}
```

Comment:
Can `Event` be just `Self` here, dropping the repeated parameter in implementations?

Reply:
Yes. Much nicer. Thanks.